test/cpp/end2end/grpclb_end2end_test.cc

```diff
@@ -593,6 +594,89 @@ TEST_F(SingleBalancerTest, InitiallyEmptyServerlist) {
   EXPECT_EQ("grpclb", channel_->GetLoadBalancingPolicyName());
 }
 
+TEST_F(SingleBalancerTest, Fallback) {
+  const int kFallbackTimeoutMs = 200 * grpc_test_slowdown_factor();
+  const int kServerlistDelayMs = 500 * grpc_test_slowdown_factor();
+  const int kCallDeadlineMs = 1000 * grpc_test_slowdown_factor();
+
+  ResetStub(kFallbackTimeoutMs);
+  std::vector<AddressData> addresses;
+  addresses.emplace_back(AddressData{balancer_servers_[0].port_, true, ""});
+  addresses.emplace_back(AddressData{balancer_servers_[1].port_, true, ""});
+  addresses.emplace_back(AddressData{balancer_servers_[2].port_, false, ""});
+  SetNextResolution(addresses);
+
+  // First response is an empty serverlist, sent right away.
+  ScheduleResponseForBalancer(0, LoadBalanceResponse(), 0);
+  // Send non-empty serverlist only after kServerlistDelayMs
+  ScheduleResponseForBalancer(
+      0, BalancerServiceImpl::BuildResponseForBackends(GetBackendPorts(), {}),
+      kServerlistDelayMs);
+
+  const gpr_timespec second_call_time = gpr_time_add(
+      gpr_now(GPR_CLOCK_REALTIME), 
+      gpr_time_from_millis(kServerlistDelayMs * 1.1, GPR_TIMESPAN));
+
+  const auto t0 = system_clock::now();
+  // The first request. The client will block while it's still trying to 
+  // contact the balancer.
+  const auto& statuses_and_responses =
+      SendRpc(kMessage_, num_backends_);
+  const auto ellapsed_ms =
+      std::chrono::duration_cast<std::chrono::milliseconds>(
+          system_clock::now() - t0);
+
+  // After fallback is in use, the call succeeds even though it's before 
+  // the serverlist is available.
+  EXPECT_GT(ellapsed_ms.count(), kFallbackTimeoutMs);
+  EXPECT_LT(ellapsed_ms.count(), kServerlistDelayMs);
+
+  // Each backend should have gotten 1 request.
```

Comment:
I don't think this is correct.  If the first RPC is sent before the serverlist is returned, it should be sent only to the backend address returned by the resolver.

If this test is currently passing, then I suspect there is a bug somewhere in either the code or the test.